specs/room_spec.rb

```diff
@@ -0,0 +1,44 @@
+require_relative 'spec_helper'
+
+describe "Room class" do
+
+  describe "#available?" do
+    it "should be available if there are no reservations" do
```

Comment:
Check these refactoring ideas out:

```ruby
describe '#available?' do
  let(:room) { Hotel::Room.new(number: 1) }

  it 'should be available if there are no reservations' do
    expect_room_to_be_available(
      Date.new(2018, 5, 1),
      Date.new(2018, 5, 5)
    )
  end

  # Organize your spec setup (before, etc) with context blocks
  context 'with a reservation' do
    let(:reservation) do
      Hotel::Reservation.new(
        start_date: Date.new(2018, 5, 1),
        end_date: Date.new(2018, 5, 5)
      )
    end

    before { room.book(reservation) }

    it 'should not be available if the start date overlaps' do
      expect_room_to_not_be_available(
        Date.new(2018, 5, 1),
        Date.new(2018, 5, 10)
      )
    end

    it 'should not be available if the end date overlaps' do
      expect_room_to_not_be_available(
        Date.new(2018, 4, 28),
        Date.new(2018, 5, 5)
      )
    end
  end

  # Create custom methods to make the specs more readable and human
  def expect_room_to_be_available(start_date, end_date)
    room.available?(start_date, end_date).must_equal true
  end

  def expect_room_to_not_be_available(start_date, end_date)
    room.available?(start_date, end_date).must_equal false
  end
end
```

Reply:
@dbalatero this is cool, much more readable. but wondering why the custom methods go at the bottom?

Reply:
@npeters5 I prefer them there so I don't have to scroll thru the custom methods just to get to the test. You can put them up top if you like though!